lang/ruby/lib/avro/io.rb

```diff
@@ -318,7 +350,7 @@ def read_fixed(writers_schema, readers_schema, decoder)
       end
 
       def read_enum(writers_schema, readers_schema, decoder)
-        index_of_symbol = decoder.read_int
+        index_of_symbol = decoder.read_enum
```

Comment:
Not really. As [per specs](https://avro.apache.org/docs/1.7.7/spec.html#binary_encoding) `An enum is encoded by a int, representing the zero-based position of the symbol in the schema.`.